packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/tests/UID.test.tsx

```diff
@@ -92,61 +102,56 @@ describe('UIDInput', () => {
       },
       required: true,
     });
-
-    await waitFor(() => expect(screen.queryByTestId('loading-wrapper')).not.toBeInTheDocument());
+    await waitForInput();
 
     expect(screen.getByRole('textbox', { name: 'Label' })).not.toHaveValue('regenerated');
   });
 
-  test('Checks the initial availability (isAvailable)', async () => {
-    jest.useFakeTimers();
-
-    render({
+  test('Checks the availability', async () => {
+    const { user } = render({
       required: true,
       initialValues: {
-        name: 'available',
+        name: 'init',
       },
     });
+    await waitForInput();
 
-    await waitFor(() => expect(screen.queryByTestId('loading-wrapper')).not.toBeInTheDocument());
+    // The value wasn't changed so the availability check should not be shown
+    expect(screen.queryByText('Available')).not.toBeInTheDocument();
+    expect(screen.queryByText('Unvailable')).not.toBeInTheDocument();
 
-    expect(screen.getByText('Available')).toBeInTheDocument();
+    jest.useFakeTimers();
+    const input = screen.getByRole('textbox');
+    await user.clear(input);
+    await user.type(input, 'available');
 
+    // Skip debouncing delay
     act(() => {
       jest.advanceTimersByTime(4000);
     });
 
-    await waitFor(() => expect(screen.queryByText('Available')).not.toBeInTheDocument(), {
-      timeout: 10000,
+    await waitFor(() => expect(screen.queryByTestId('loading-wrapper')).not.toBeInTheDocument());
+    expect(await screen.findByText(/^Available$/)).toBeInTheDocument();
+
+    // Change the value to make it unavailable
+    await user.type(input, 'unavailable');
+    act(() => {
+      jest.advanceTimersByTime(4000);
     });
+    expect(await screen.findByText(/^Unavailable$/)).toBeInTheDocument();
```

Comment:
Could you please use different uid than `available` and `unavailable`, I am not sure why your unavailability check is on here?

Reply:
I can see how the uid is confusing so I updated it.

Regarding the check itself, I'm testing the general availability check behavior, so it makes sense to me to check both outcomes, the positive and negative results